src/client/particles.cpp

```diff
@@ -577,30 +684,31 @@ void ParticleManager::addNodeParticle(IGameDef *gamedef,
 	LocalPlayer *player, v3s16 pos, const MapNode &n, const ContentFeatures &f)
 {
 	ParticleParameters p;
-	video::ITexture *texture;
+	ClientParticleTexture texture;
+	texture.ref = nullptr;
```

Comment:
Looks like a common pattern, you can avoid this by initializing `ref` in the `ClientParticleTexture` ctor